libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthenticationFactory.cs

```diff
@@ -43,18 +44,18 @@ public static class BotFrameworkAuthenticationFactory
             HttpClient httpClient,
             ILogger logger)
         {
-            if (string.IsNullOrEmpty(channelService))
-            {
-                return new PublicCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
-            }
-            else if (channelService == GovernmentAuthenticationConstants.ChannelService)
-            {
-                return new GovernmentCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
-            }
-            else
+            if (
+                !string.IsNullOrEmpty(toChannelFromBotLoginUrl) || 
+                !string.IsNullOrEmpty(toChannelFromBotOAuthScope) || 
+                !string.IsNullOrEmpty(toBotFromChannelTokenIssuer) || 
+                !string.IsNullOrEmpty(oAuthUrl) ||
+                !string.IsNullOrEmpty(toBotFromChannelOpenIdMetadataUrl) ||
+                !string.IsNullOrEmpty(toBotFromEmulatorOpenIdMetadataUrl) ||
+                !string.IsNullOrEmpty(callerId))
```

Comment:
@willportnoy is the `callerId` required? We're following the activity spec and discard of any callerIds received on the wire. What is this `callerId` used for, if it's not generated from the SDK post-successful authentication of incoming activities?

Reply:
Based on https://github.com/microsoft/botbuilder-dotnet/commit/5efd3d0c04d041f847ca8cdfa5f61c522192d047 I would guess it's related to skills.  Maybe @gabog can comment?